linkerd/stack/src/gate.rs

```diff
@@ -262,6 +264,47 @@ mod tests {
         assert_ready!(gate.poll_ready()).expect("ok");
     }
 
+    #[tokio::test]
+    async fn gate_repolls_back_to_pending() {
```

Comment:
i think this behavior is complex enough that it would be nice to have a comment here explaining what specific properties we are testing for here?